### PR TITLE
#181339536 Reject/Approve Request (By Requester’s Manager)

### PR DESCRIPTION
**Pull request by niyonkurueric**
### what does this Pull Request do?
this Pull Request implements how managers approve and reject the trip request
### description of tasks to be completed

- user should be able to send a request 
- the manager should be able to approve and reject requests of the user
- manager retrieves all Trip Request owner 
### how should this Pull Request be tested?

- locally: clone this repo and checkout **feature/#181339536-reject-approve-request**  branch
-   remottelly: **https://staging-bare-feature-18-vovbmt.herokuapp.com**
 - add all environment variables and run yarn run dev after running yarn run dbmigrate:all to seed data
 - checkout this api to login
 
            - **POST**  http://localhost:5000/api/v1/users/login

 - checkout this API to get all assigned users to the manager
 
             - **PUT**  http://localhost:5000/api/v1/users/assign-to-manager

- get the token and authenticate with a bearer and then hit these APIs to test the Approve and reject the trip request

           - **PATCH**  http://localhost:5000/api/v1/trips/{id}
### What are the relevant pivotal tracker stories?
[#181339536](https://www.pivotaltracker.com/story/show/181339536)
### Screenshots (if appropriate)
Screenshot from 2022-03-23 23-13-40

### CODE REVIEW

**Review by Code-250 — APPROVED** [review]
please fix your linter

**Review by Gabin-ishimwe — APPROVED** [review]

File: src/controllers/trip.controller.js [hunk 1]
```diff
@@ -6,85 +6,100 @@ import { getQuery } from '../helpers/trip.helpers';
...
-        return res.status(400).json({
-          message: 'Access denied, You are not assigned a manager'
-        });
-      }
+    static async createTrip(req, res) {
+        try {
+            if (!req.user.manager_id) {
+                return res.status(400).json({
+                    message: 'Access denied, You are not assigned a manager'
+                });
```
Comment (ncutixavier) [inline src/controllers/trip.controller.js:37]: Change status code

File: src/middlewares/role.middleware.js [hunk 2]
```diff
@@ -6,90 +6,88 @@ import { User, Blacklist } from '../database/models';
...
 
-  req.user = { email, id, manager_id };
-  next();
+    req.user = { email, id, manager_id };
+    next();
 };
 
-export const checkRoleSame = async (req, res, next) => {
-  const { email, role } = req.body;
+export const checkRoleSame = async(req, res, next) => {
```
Comment (ncutixavier) [inline src/middlewares/role.middleware.js:39]: What about this middleware? Try to use meaningful name and make sure that you're returning appropriate status code

File: tests/password-reset.test.js [hunk 3]
```diff
@@ -7,69 +7,69 @@ import { generateToken } from '../src/helpers/user.helpers';
...
+    // it('should Send password reset link', (done) => {
+    //   chai
+    //     .request(app)
+    //     .post(`/api/v1/users/forgot-password`)
+    //     .send({ email: 'SUPER_ADMIN@gmail.com' })
+    //     .end((err, res) => {
+    //       expect(res).to.have.status([200]);
+    //       done();
+    //     });
+    // });
```
Comment (ncutixavier) [inline tests/password-reset.test.js]: Remove commented code

**Review by ncutixavier — CHANGES_REQUESTED** [review]

**Review by Tsa2341 — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**